### ISSUE TITLE
wxpg.PGEditor.GetValueFromControl & wxpg.PGProperty.StringToValue

### ISSUE THREAD

**Issue by Lokla** [wxWidgets/Phoenix#742]
**Operating system**: Windows 10 64 Bit
**wxPython version**: 4.0.2a1 msw (phoenix)
**Stock or custom build**: Stock
**Python version**: 2.7.14
**Stock or custom build**: Stock

Related with #394 
Reported in wxpython-users here: 
https://groups.google.com/forum/#!topic/wxpython-users/F5l-QrKQEfk

I have two problems here when implementing a custom editor. I tried to fix it myself, but I am stuck along the way.

**PGEditor Overload of GetValueFromControl crashes application**:

File to create the crash:
[pgEditorCrash.zip](https://github.com/wxWidgets/Phoenix/files/1710135/pgEditorCrash.zip)

Also happens in demo (example is created mostly from demo code).

Steps to reproduce:
 * Edit the label
 * Save the edited value

I already managed to at least jump into the function by removing the third D in the format - I think it might be too much there. But I have no experience with sip, so it is just a guess.

```python
PyObject *sipResObj = sipCallMethod(0, sipMethod, "DD",
                                            property, sipType_wxPGProperty, NULL,
                                            ctrl, sipType_wxWindow, NULL);
```

My code now reaches the Python function (yay), but the return value (once the second problem is avoided) will still report an error:

```TypeError: invalid result from CustomEditor.GetValueFromControl(), a 'bool' is expected not 'tuple'```

So it seems that the sipParseResult is not executed correctly.

But I also ran into the following additional problem:

**PGProperty function StringToValue returns boolean and not tuple of Boolean and String**:

```python
import wx
import wx.propgrid as wxpg
app = wx.App()
frame = wx.Frame(None)
pg = wxpg.PropertyGrid(frame)
prop = wxpg.StringProperty("Simple Property", name="mine", value="Change me!!")
print(prop.StringToValue(wxpg.PG_EDITABLE_VALUE, 'My New Value'))
```

* This code returns just the boolean True which indicates that the result is valid, but not the result itself.
* At the moment this function also cannot be overridden for custom Properties

This might also affect IntToValue.

For funtions like the GetValueFromControl, you really need both return values (the bool and the string), so the function should return a tuple.

Thanks,
Michael

**Comment by jensgoe** [comment]
@RobinD42 
I can confirm this crash. It happens with all custom editors that overrides GetValueFromControl. Somehow sip does not handle this method correctly. 

To reproduce:
* start the wxPython demo (PropertyGrid.py)
* start editing and CHANGE the value of the last editor (StringWithCustomEditor) 
* confirm the changes by clicking somewhere or hit enter
* --> Crash

The debugger stops in generated code of sip.

**Comment by jensgoe** [comment]
Ok, i fixed the crash by removing the virtualCatcherCode of GetValueFromControl in the etg script. After that i had the same issue like Lokla that StringToValue is not working properly. All the StringToValue functions that receives the result in the first argument are not marked as an out parameter in the sip code. I fixed this by modifying the sip file like this (adding /Out/):

```
    virtual
    bool StringToValue(
        wxPGVariant & variant   /Out/,
        const wxString & text,
        int argFlags = 0
    ) const;
```

This seems to be done in the etg script by:
```
    c = module.find('wxPGProperty')
    tools.ignoreConstOverloads(c)
    c.find('StringToValue.variant').out = True
    c.find('IntToValue.variant').out = True
```
Somehow this does not work. I guess this seems not to consider all the derived subclasses.

**Comment by RobinD42** [comment]
Fixed by #877
